hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -182,11 +182,6 @@ private[hudi] object HoodieSparkSqlWriter {
             } else {
               hoodieAllIncomingRecords
             }
-
-          if (hoodieRecords.isEmpty()) {
```

Comment:
Have you done any experiments on this? Like putting another Spark action before this. 
`rdd.isEmpty()` is the most efficient way I could find that will do the empty check. Maybe we can change the description of this action so the user won't misunderstand?